src/bluehawk.ts

```diff
@@ -0,0 +1,60 @@
+import { makeBlockCommentTokens } from "./lexer/makeBlockCommentTokens";
+import { makeLineCommentToken } from "./lexer/makeLineCommentToken";
+import { makeCstVisitor } from "./parser/makeCstVisitor";
+import { validateVisitorResult } from "./parser/validator";
+import { RootParser } from "./parser/RootParser";
+
+export interface Location {
+  line: number;
+  column: number;
+  offset: number;
+}
+
+export interface Range {
+  start: Location;
+  end: Location;
+}
+
+export interface BluehawkError {
+  message: string;
+  location: Location;
+}
+
+export class BluehawkResult {
+  errors: BluehawkError[];
+}
+
+export class BluehawkSource {
+  text: string;
+  language: string;
+}
+
+export class Bluehawk {
+  parsers = new Map<string, RootParser>();
+
+  run(source: BluehawkSource): BluehawkResult {
+    if (!this.parsers.has(source.language)) {
+      this.parsers.set(
+        source.language,
+        new RootParser([
+          // TODO: map source.language to block/line comment tokens
+          ...makeBlockCommentTokens(/\/\*/y, /\*\//y),
+          makeLineCommentToken(/\/\//y),
+        ])
+      );
+    }
+    const parser = this.parsers.get(source.language);
+    const { lexer } = parser;
```

Comment:
Why this?

Reply:
copy/pasted silliness i kept meaning to remove. Fixed.